pkg/doc/jwt/claims.go

```diff
@@ -0,0 +1,123 @@
+/*
+ *
+ *  * Copyright SecureKey Technologies Inc. All Rights Reserved.
+ *  *
+ *  * SPDX-License-Identifier: Apache-2.0
+ *
+ */
+
+package jwt
+
+import (
+	"encoding/json"
+	"errors"
+	"strconv"
+	"time"
+)
+
+// Claims defines JWT Claims Set (https://tools.ietf.org/html/rfc7519#section-4)
+type Claims struct {
+
+	// Issuer defines iss claim.
+	Issuer string `json:"iss,omitempty"`
+
+	// Subject defines sub claim.
+	Subject string `json:"sub,omitempty"`
+
+	// Audience defines aud claim.
+	Audience Audience `json:"aud,omitempty"`
+
+	// Expiry defines exp claim.
+	Expiry *NumericDate `json:"exp,omitempty"`
+
+	// NotBefore defines nbf claim.
+	NotBefore *NumericDate `json:"nbf,omitempty"`
+
+	// IssuedAt defines iat claim.
+	IssuedAt *NumericDate `json:"iat,omitempty"`
+
+	// ID defines jti claim.
+	ID string `json:"jti,omitempty"`
+}
+
+// NumericDate is a JSON numeric value representing the number of seconds from
+// 1970-01-01T00:00:00Z UTC until the specified UTC date/time, ignoring leap seconds.
+type NumericDate int64
+
+// NewNumericDate creates a new NumericDate from time.Time
+func NewNumericDate(t time.Time) *NumericDate {
+	if t.IsZero() {
+		return nil
+	}
+
+	nd := NumericDate(t.Unix())
+
+	return &nd
+}
+
+// MarshalJSON converts NumericDate to JSON bytes.
+func (n NumericDate) MarshalJSON() ([]byte, error) {
+	return []byte(strconv.FormatInt(int64(n), 10)), nil
+}
+
+// UnmarshalJSON defines custom unmarshalling of NumericDate from JSON bytes.
+func (n *NumericDate) UnmarshalJSON(data []byte) error {
```

Comment:
this seems like a constructor rather than a member method.

Reply:
It implements `json.Unmarshaler` interface so that custom `NumericDate` unmarshalling is enabled.

Reply:
I feel like we shouldn't need to export these marshal/unmarshal methods. i.e., in other packages we have the raw pattern to enable us to hide such details.